test/Npgsql.Tests/AutoPrepareTests.cs

```diff
@@ -501,6 +500,58 @@ public void ReplaceWithBadSql()
 
             Assert.That(conn.ExecuteScalar("SELECT 2"), Is.EqualTo(2));
         }
+        
+        [Test, IssueLink("https://github.com/npgsql/npgsql/issues/4082")]
+        public void PreparedStatementAlreadyExistsFailure()
+        {
+            var csb = new NpgsqlConnectionStringBuilder(ConnectionString)
```

Comment:
Better to use CreateTempPool, which creates a unique connection pool which gets cleared at the end of the test. This ensures two tests don't accidentally use the same pool (and interfere with each other), and also gets rid of the physical connection (the test's single connection currently sits idle until the end of the entire test suite).

Reply:
Thanks for that. Updated now.